src/Magnum/Math/Quaternion.h

```diff
@@ -733,6 +740,57 @@ template<class T> Matrix3x3<T> Quaternion<T>::toMatrix() const {
     };
 }
 
+/* Algorithm from:
+   https://github.com/qt/qtbase/blob/84056cdca11c63ddb9fac3d95fcfa952adaa6ea1/src/gui/math3d/qquaternion.cpp#L507 */
+template<class T> Vector3<T> Quaternion<T>::toRotation() const {
+    Vector3<T> euler;
+
+    T xp = vector().x();
+    T yp = vector().y();
+    T zp = vector().z();
+    T wp = scalar();
+
+    T xx = xp*xp;
+    T xy = xp*yp;
+    T xz = xp*zp;
+    T xw = xp*wp;
+    T yy = yp*yp;
+    T yz = yp*zp;
+    T yw = yp*wp;
+    T zz = zp*zp;
+    T zw = zp*wp;
+
+    const T lengthSquared = xx + yy + zz + wp*wp;
+    if(!TypeTraits<T>::equalsZero(lengthSquared - 1.0f, 100) &&
+        !TypeTraits<T>::equalsZero(lengthSquared, 100)) {
```

Comment:
Not sure about the `100` here -- doesn't it make the comparison *too* fuzzy? If the quat is normalized (which is a prerequisite for extracting the angles?), the original magnitude is `T(1.0)`, so that's what should be there I think.